### ISSUE TITLE
Create ability to catch errors

### ISSUE THREAD

**Issue by asiniy** [redux-json-api/redux-json-api#46]
In both master branch and axios implementation https://github.com/dixieio/redux-json-api/pull/41 there is no common way how to handle errors, like that:

``` javascript
setAxiosPromiseHandler({
  then: [(result) => { console.log(result) }],
  catch: [(error) => { console.log(error) }],
})
```

This is extremely helpful feature in things like global authentication. Anywhere when user gets `Forbidden`, he is redirected to sign in page for example. No need to write same exception handlers anywhere.

What do you think?

**Comment by gavmck** [comment]
Any movement on this one?

**Comment by asiniy** [comment]
@gavmck I forgot `redux` and `react` as nightmare so nothing from my side. Try to play with promises, `redux-json-api` is easy to understand and thus change

**Comment by egeriis** [comment]
@gavmck I think it has been missed. I can definitely see the use case, so I think it should be implemented for v2.0—which is a bit overdue to be honest. Do you have the chance to create a PR?

**Comment by dakota** [comment]
Closing due to no movement.

You can accomplish the same thing by using [axios interceptors](https://github.com/axios/axios#interceptors).

```js
axios.interceptors.response.use(
  (response) => response,
  function (error) {
    if (axios.isCancel(error)) {
      return;
    }

    if (error.response.status === 401 || error.response.status === 403) {
      window.location.reload(true);
    }

    return Promise.reject(error.response);
  }
);
```
